### PR TITLE
luci-app-aria2: Rewrite all

### PR DESCRIPTION
**Pull request by kuoruan**
This package depends on https://github.com/openwrt/packages/pull/4701 (actually not).

Signed-off-by: Hsing-wang Liao <kuoruan@gmail.com>

### CODE REVIEW

File: applications/luci-app-aria2/Makefile [hunk 1]
```diff
@@ -1,31 +1,22 @@
 #
-# Copyright (C) 2016 Openwrt.org
+# Copyright (C) 2016-2017 Openwrt.org
```
Comment (jow-) [inline applications/luci-app-aria2/Makefile]: Leave the original copyright as is and add your own.

**Comment by kuoruan** [comment]
@Ansuel Add new options and updated aria2 package. I did a simple test. Please help me check.

**Comment by Ansuel** [comment]
@kuoruan compiling right now

**Comment by kuoruan** [comment]
@Ansuel Seems every thing OK, pls do a check., Thanks.

**Comment by Ansuel** [comment]
@kuoruan sorry for annoying that much...
another problem... custom option are not displayed in luci app (still present in config file) (aria 2 starts so that problem is fixed)

**Comment by kuoruan** [comment]
@Ansuel Use ```extra_setting``` not ```extra_settings``` in new version.

**Comment by Ansuel** [comment]
ok... now i have problem with rpc secure... don't know if it's me or not... with old luci-app and the extra settings was working fine... don't know... 
i have this in log
Exception: [HttpHeaderProcessor.cc:165] errorCode=1 Bad Request-Line: missing request-target
2017-08-14 11:29:51.820840 [INFO] [HttpServerCommand.cc:275] CUID#37 - Error occurred while reading HTTP request
2017-08-14 11:29:51.820660 [INFO] [HttpListenCommand.cc:78] RPC: Accepted the connection from 192.168.2.11:1087.

i don't think they are related to your package. everything else is good

**Comment by kuoruan** [comment]
@Ansuel Maybe you should have a look at the config file generate by my init script.
And find out if this problem is caused by my package.(Lost some options?)

**Comment by Ansuel** [comment]
where is generated the aria2 config file (too lazy to check the script lol)

**Comment by kuoruan** [comment]
@Ansuel ```/var/etc/aria2/aria2.conf.main``` by default. OK, I will add a new tab to show the config file .

**Comment by Ansuel** [comment]
@kuoruan problem! RPC CERTIFICATE doesn't appear in config main

**Comment by kuoruan** [comment]
Did you update the aria2 init script, It's OK in my device.

**Comment by Ansuel** [comment]
@kuoruan such a bad tester... ok i updated the init.d script now all is good!
something to test?

**Comment by kuoruan** [comment]
@Ansuel I am adding the "Files" tag. I will notice you when I finish. Thank you very much.

**Comment by Ansuel** [comment]
also position of session and insertion file would be good (so in case of a restart the file can be restared automatically)

**Comment by kuoruan** [comment]
@Ansuel Added a new tab to show config file and session file.
But what is "insertion file"?

**Comment by Ansuel** [comment]
@kuoruan Save error/unfinished downloads to FILE on exit. You can pass this output file to aria2c with --input-file option on restart.

**Comment by kuoruan** [comment]
@Ansuel Do you find any other problems? Like everything works well.

**Comment by kuoruan** [comment]
@jow- Done. please check.

**Comment by Ansuel** [comment]
@kuoruan all good :)

**Comment by Ansuel** [comment]
No merge?

**Comment by kuoruan** [comment]
@Ansuel Maybe I need to open a new pull request.

**Comment by Ansuel** [comment]
@kuoruan nha... @jow- can you merge this?

**Comment by Ansuel** [comment]
@kuoruan solve conflicts pls

**Comment by kuoruan** [comment]
@Ansuel I'm adding user option, wait a moment.

**Comment by kuoruan** [comment]
@Ansuel Done.

**Comment by kuoruan** [comment]
@yousong Pls help me review this PR.

**Comment by yousong** [comment]
tl;nd ;)

Please describe in the commit message body why the rewrite is needed and what has changed.  If the rewrite will invalidate all the existing translations, how about just delete them and add them back at a later time so that the diffstat can be less overwhelming...

**Comment by kuoruan** [comment]
@yousong 英语太难写了，所以我就没写详细介绍，等我慢慢翻译翻译，再加上。
最近写 luci 写得比较熟了，看着以前写的这个不爽就重写了，你先帮我看看有没有错误的地方。
143925vqicqbqden1w7e40

**Comment by Ansuel** [comment]
Why no merge :(

**Comment by Ansuel** [comment]
@kuoruan why closed :(((

**Pull request closed without merging**